media/transcoder/src/main/java/com/example/transcoder/CreateJobFromPresetBatchMode.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.transcoder;
+
+// [START transcoder_create_job_from_preset_batch_mode]
+
+import com.google.cloud.video.transcoder.v1.CreateJobRequest;
+import com.google.cloud.video.transcoder.v1.Job;
+import com.google.cloud.video.transcoder.v1.Job.ProcessingMode;
+import com.google.cloud.video.transcoder.v1.LocationName;
+import com.google.cloud.video.transcoder.v1.TranscoderServiceClient;
+import java.io.IOException;
+
+public class CreateJobFromPresetBatchMode {
+
+  public static void main(String[] args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "my-project-id";
+    String location = "us-central1";
+    String inputUri = "gs://my-bucket/my-video-file";
+    String outputUri = "gs://my-bucket/my-output-folder/";
+    String preset = "preset/web-hd";
+
+    createJobFromPresetBatchMode(projectId, location, inputUri, outputUri, preset);
+  }
+
+  // Creates a job from a preset in batch mode.
+  public static void createJobFromPresetBatchMode(
+      String projectId, String location, String inputUri, String outputUri, String preset)
+      throws IOException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests.
+    try (TranscoderServiceClient transcoderServiceClient = TranscoderServiceClient.create()) {
+
+      var createJobRequest =
+          CreateJobRequest.newBuilder()
+              .setJob(
+                  Job.newBuilder()
+                      .setInputUri(inputUri)
+                      .setOutputUri(outputUri)
+                      .setTemplateId(preset)
+                      .setMode(ProcessingMode.PROCESSING_MODE_BATCH)
+                      .setBatchModePriority(10)
+                      .build())
+              .setParent(LocationName.of(projectId, location).toString())
+              .build();
+
+      // Send the job creation request and process the response.
+      Job job = transcoderServiceClient.createJob(createJobRequest);
```

Comment:
Can we block and wait until the job is completed?

Reply:
Unfortunately, no. The API is asynchronous. You have to poll the job state. There is no user-exposed LRO.